builder/dockerfile/dispatchers.go

```diff
@@ -210,20 +210,39 @@ func (d *dispatchRequest) getExpandedImageName(shlex *shell.Lex, name string) (s
 	}
 	return name, nil
 }
-func (d *dispatchRequest) getImageOrStage(name string) (builder.Image, error) {
+
+// getOsFromFlagsAndStage calculates the operating system if we need to pull an image.
+// stagePlatform contains the value supplied by optional `--platform=` on
+// a current FROM statement. b.builder.options.Platform contains the operating
+// system part of the optional flag passed in the API call (or CLI flag
+// through `docker build --platform=...`). Precedence is for an explicit
+// platform indication in the FROM statement.
+func (d *dispatchRequest) getOsFromFlagsAndStage(stagePlatform string) string {
+	switch {
+	case stagePlatform != "":
+		return stagePlatform
+	case d.builder.options.Platform != "":
+		return d.builder.options.Platform
+	default:
+		return runtime.GOOS
+	}
+}
+
+func (d *dispatchRequest) getImageOrStage(name string, stagePlatform string) (builder.Image, error) {
```

Comment:
This variable will only contain `OS` right? should be renamed to avoid confusion, or use the full platform struct.

Reply:
Fair. Update being pushed shortly.